src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,99 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Entry<K, V>[] table;
+    private int size;
+    private int threshold;
+    private final float loadFactor;
+
+    static class Entry<K, V> {
+        final K key;
+        V value;
+        Entry<K, V> next;
+
+        public Entry(K key, V value, Entry<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    public MyHashMap() {
+        this(DEFAULT_INITIAL_CAPACITY, DEFAULT_LOAD_FACTOR);
+    }
+
+    public MyHashMap(int initialCapacity, float loadFactor) {
+        this.loadFactor = loadFactor;
+        this.threshold = (int) (initialCapacity * loadFactor);
+        this.table = new Entry[initialCapacity];
+    }
 
     @Override
     public void put(K key, V value) {
+        int hash = hash(key);
+        int index = indexFor(hash, table.length);
 
-    }
+        for (Entry<K, V> entry = table[index]; entry != null; entry = entry.next) {
+            if ((entry.key == null && key == null) || (entry.key != null && entry.key.equals(key))) {
+                entry.value = value;
+                return;
+            }
+        }
+        addEntry(hash, key, value, index);
+        }
 
     @Override
-    public V getValue(K key) {
+    public V getValue (K key){
+        int hash = hash(key);
+        int index = indexFor(hash, table.length);
+
+        for (Entry<K, V> entry = table[index]; entry != null; entry = entry.next) {
+            if ((entry.key == null && key == null) || (entry.key != null && entry.key.equals(key))) {
+                return entry.value;
+            }
+        }
         return null;
     }
 
     @Override
-    public int getSize() {
-        return 0;
+    public int getSize () {
+        return size;
+    }
+
+    private int indexFor ( int hash, int length){
+        return hash & (length - 1);
+    }
+
+    private int hash (Object key){
+        return key == null ? 0 : key.hashCode() ^ (key.hashCode() >>> 16);
+    }
+
+    private void addEntry(int hash, K key, V value, int index) {
+        Entry<K, V> newEntry = new Entry<>(key, value, table[index]);
+        table[index] = newEntry;
+        if (++size >= threshold) {
+            resize(2 * table.length);
+        }
+    }
+
+    private void resize(int newCapacity) {
+        Entry<K, V>[] oldTable = table;
+        table = new Entry[newCapacity];
```

Comment:
Be attentive with the diamond operator in Java. It should be parameterized with the correct types to avoid unchecked assignment warnings. Use 'new Entry<K, V>[newCapacity]' instead of 'new Entry[newCapacity]'. Note that in Java, generic array creation is not allowed, so you will need to use a cast or an alternative approach such as an ArrayList.